src/settingsScreen.js

```diff
@@ -38,53 +48,336 @@ class SettingsScreen {
+        await match.accessible.Label(this.language.settingsScreen.SaveToSimulator).tap();
+    }
+    async Apply(limits) {
+        await match.UIEditableTextField().atIndex(0).clearText();
+        await match.UIEditableTextField().atIndex(0).typeText(String(limits.maxBasalRate));
+        await match.UIEditableTextField().atIndex(0).tapReturnKey();
+        await expect(match.UIEditableTextField().atIndex(0)).toHaveText(String(limits.maxBasalRate));
+        await match.UIEditableTextField().atIndex(1).clearText();
+        await match.UIEditableTextField().atIndex(1).typeText(String(limits.maxBolus));
+        await match.UIEditableTextField().atIndex(1).tapReturnKey();
+        await expect(match.UIEditableTextField().atIndex(1)).toHaveText(String(limits.maxBolus));
+    }
+    async ApplyWithExpectations(limits, additionalExpectations) {
+        await this.Apply(limits);
+        if (additionalExpectations) {
+            await additionalExpectations();
+        }
+    }
+}
+
+class InsulinSensitivitiesScreen {
+    constructor(language) {
+        this.language = language;
+    }
+    InsulinSensitivitiesHeader() {
+        return match.accessible.Header(this.language.settingsScreen.InsulinSensitivities);
+    }
+    async Cancel() {
+        await match.accessible.ButtonBarButton(this.language.general.Cancel).tap();
+    }
+    async Close() {
+        await match.accessible.BackButton(this.language.settingsScreen.Settings).tap();
+    }
+    async Save() {
+        await match.accessible.Label(this.language.general.Save).tap();
+    }
+    async ApplyAll(sensitivities) {
+        if (sensitivities) {
+            for (let index = 0; index < sensitivities.length; index++) {
+                await this.Apply(sensitivities[index]);
+            }
+        }
+    }
+    /**
+     * @param {Object} sensitivity
+     * @param {String} sensitivity.time
+     * @param {String} sensitivity.bgValuePerInsulinUnit
+     */
+    async Apply(sensitivity) {
+        await match.accessible.ButtonBarButton(this.language.general.Add).tap();
+        //select time unless this is the first Insulin Sensitivitiy we have set...
+        if (sensitivity.time != "12:00 AM") {
+            await match.accessible.Label(`${sensitivity.time}`).atIndex(0).tap();
+        }
+        await match.accessible.SetPickerValue(1, `${sensitivity.bgValuePerInsulinUnit} ${config.insulinSensitivitiesUnits}`);
+    }
+    /**
+     * @param {Object} sensitivity
+     * @param {String} sensitivity.time
+     * @param {String} sensitivity.bgValuePerInsulinUnit
+     */
+    async Edit(sensitivity) {
+        await match.accessible.Label(`${sensitivity.time}`).atIndex(0).tap();
+        await match.accessible.SetPickerValue(1, `${sensitivity.bgValuePerInsulinUnit} ${config.insulinSensitivitiesUnits}`);
+    }
+}
+
+class CorrectionRangeScreen {
+    constructor(language) {
+        this.language = language;
+    }
+    _pickerIndexForTime(time) {
+        switch (time) {
```

Comment:
[nit] Consider just building an array of all 24 times and then doing a find to get the index? Also, should the times be in the "language" file for localization and 24-hour clock?

Reply:
Perhaps the array could just be in the language file?

Reply:
true - will move